amads/harmony/root_finding/parncutt.py

```diff
@@ -0,0 +1,190 @@
+        self.root_support_weights = self.load_root_support_weights(root_support_weights)
+        self.exponent = exponent
+        self.root_strengths = [self.get_root_strength(pc) for pc in range(12)]
+        self.root = self.get_root()
+        self.root_ambiguity = self.get_root_ambiguity()
+
+    def load_root_support_weights(
+        self, root_support_weights: Union[str, Dict[int, float]]
+    ) -> Dict[int, float]:
+        if isinstance(root_support_weights, str):
+            if root_support_weights not in self.available_root_support_weights:
+                raise ValueError(
+                    f"Unknown root support weights version: {root_support_weights}"
+                )
+        return self.available_root_support_weights[root_support_weights]
+
+    def get_root_strength(self, pc: int) -> float:
+        return sum(
+            support_weight
+            for interval, support_weight in self.root_support_weights.items()
+            if (pc + interval) % 12 in self.pc_set
+        )
+
+    def get_root(self) -> int:
+        return self.root_strengths.index(max(self.root_strengths))
+
+    def get_root_ambiguity(self) -> float:
+        max_weight = max(self.root_strengths)
+        if max_weight == 0:
+            return 0.0
+        return sum(w / max_weight for w in self.root_strengths) ** self.exponent
+
+    def visualize(self, title: Optional[str] = None) -> None:
+        """
+        Visualize the root support weights for a chord.
+
+        Parameters
+        ----------
+        title : Optional[str], optional
+            Title for the plot, by default None.
+
+        Examples
+        --------
+        >>> analysis = ParncuttRootAnalysis([0, 4, 7])
+        >>> analysis.visualize()
+        """
+        # Create a bar chart
+        plt.figure(figsize=(10, 6))
+        bars = plt.bar(range(12), self.root_strengths)
+
+        # Highlight the root
+        bars[self.root].set_color("red")
+
+        # Add labels
+        plt.xlabel("Pitch class")
+        plt.ylabel("Root strength")
+
+        # Add pitch class names
+        pitch_class_names = [
+            "C",
+            "C#",
+            "D",
+            "D#",
+            "E",
+            "F",
+            "F#",
+            "G",
+            "G#",
+            "A",
+            "A#",
+            "B",
+        ]
+        plt.xticks(range(12), pitch_class_names)
+
+        # Add title
+        if title is None:
+            chord_str = ", ".join(
+                str(pc) for pc in sorted(set(p % 12 for p in self.chord))
+            )
+            title = f"Root strengths for chord [{chord_str}]"
+        plt.title(title)
+
+        # Add grid
+        plt.grid(axis="y", linestyle="--", alpha=0.7)
+
+        plt.tight_layout()
+
+        # Check if we're in a Pytest environment
+        if os.environ.get("PYTEST_CURRENT_TEST"):
+            plt.close()  # Close the figure to prevent memory leaks
+        else:
+            plt.show()
```

Comment:
Need a repo-wide policy on this

Reply:
Good thoughts about this. I looked at what Roger has done elsewhere, and saw that he has a nice solution where he returns `plt, fig` from the function, so I've now adopted that for consistency.